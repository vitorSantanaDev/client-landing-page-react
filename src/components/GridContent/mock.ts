export default {
  title: 'Title text example',
  content: `<p>Lorem ipsum dolor sit amet, <a href="#">consectetur adipiscing</a> elit. Sed ac lacus pretium urna malesuada dignissim. Sed risus tortor, sagittis eget sem ut, porta imperdiet magna. In fringilla nisi enim. Aenean mollis vehicula nulla, non vulputate eros pretium eget. Vivamus tincidunt accumsan massa non convallis. Aliquam molestie efficitur faucibus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Vestibulum suscipit convallis mi, vitae ornare dolor hendrerit quis. Nam gravida orci posuere nisi imperdiet posuere. Phasellus quis blandit turpis, eu feugiat velit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Integer ac malesuada quam. Aliquam eu finibus massa.</p>

  <p>Donec pulvinar volutpat ante quis porttitor. In hac habitasse platea dictumst. Sed eu placerat turpis, vitae pulvinar odio. Pellentesque viverra posuere enim, eu dapibus quam congue sit amet. Sed iaculis eleifend scelerisque. Curabitur ut elit at sem cursus euismod. Praesent ornare, urna vitae sollicitudin tincidunt, sem odio accumsan velit, quis iaculis neque elit vel ante. Suspendisse rhoncus felis at arcu blandit facilisis. Cras euismod varius nibh, a cursus nisi. Integer laoreet posuere sem vel pulvinar. Etiam vel blandit libero. Praesent consectetur tellus et justo lobortis rhoncus.</p>
  `,
  background: false
}
