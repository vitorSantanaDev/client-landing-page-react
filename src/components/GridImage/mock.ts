export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      image: {
        altText: 'Teste 1',
        srcImage: 'https://source.unsplash.com/random/800x800?r=1'
      }
    },
    {
      image: {
        altText: 'Teste 2',
        srcImage: 'https://source.unsplash.com/random/800x800?r=2'
      }
    },
    {
      image: {
        altText: 'Teste 3',
        srcImage: 'https://source.unsplash.com/random/800x800?r=3'
      }
    },
    {
      image: {
        altText: 'Teste 4',
        srcImage: 'https://source.unsplash.com/random/800x800?r=4'
      }
    },
    {
      image: {
        altText: 'Teste 5',
        srcImage: 'https://source.unsplash.com/random/800x800?r=5'
      }
    },
    {
      image: {
        altText: 'Teste 6',
        srcImage: 'https://source.unsplash.com/random/800x800?r=6'
      }
    }
  ]
}
