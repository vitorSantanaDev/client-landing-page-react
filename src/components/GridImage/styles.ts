import styled, { css } from 'styled-components'
import { Wrapper as TextComponent } from '../TextComponent/styles'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`
export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    gap: ${theme.spacings.large};
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  `}
`
export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`
export const Image = styled.img`
  ${() => css`
    width: 100%;
    max-height: 447px;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`
