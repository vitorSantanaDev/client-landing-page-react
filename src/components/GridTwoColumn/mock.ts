export default {
  title: 'Grid two column',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ad vero, accusantium eum exercitationem in ex quo veniam sed totam, quos ipsam at. Facilis a labore alias sed, natus dolor temporibus?',
  srcImage: 'assets/images/javascript.svg'
}
