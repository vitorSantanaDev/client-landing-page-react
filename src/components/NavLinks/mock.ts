export default [
  { children: 'First Link', link: '#target_1', newTab: false },
  { children: 'Second Link', link: '#target_2', newTab: true },
  { children: 'Tertiary Link', link: '#target_3', newTab: false },
  { children: 'Last Link', link: '#target_4', newTab: true },
  { children: 'Test Link', link: '#target_5', newTab: false },
  { children: 'Contact Link', link: '#target_6', newTab: false },
  { children: 'About Link', link: '#target_7', newTab: true }
]
