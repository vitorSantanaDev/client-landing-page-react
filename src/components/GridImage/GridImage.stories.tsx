import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import GridImage from '.'
import mock from './mock'

export default {
  title: 'GridImage',
  component: GridImage,
  args: { ...mock }
} as ComponentMeta<typeof GridImage>

export const Default: ComponentStory<typeof GridImage> = (args) => (
  <GridImage {...args} />
)
