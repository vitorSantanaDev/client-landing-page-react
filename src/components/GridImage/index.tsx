import React from 'react'
import Heading from '../Heading'
import SectionBackground from '../SectionBackground'
import TextComponent from '../TextComponent'
import * as S from './styles'

interface IGridImage {
  background?: boolean
  title: string
  description: string
  grid: {
    altText: string
    srcImage: string
  }[]
}

const GridImage: React.FC<IGridImage> = ({
  title,
  background,
  description,
  grid
}) => (
  <SectionBackground background={background}>
    <S.Wrapper>
      <Heading size="huge" uppercase colorDark={!background} as="h2">
        {title}
      </Heading>
      <TextComponent>{description}</TextComponent>
      <S.Grid>
        {grid.map((element) => (
          <S.GridElement key={element.srcImage}>
            <S.Image src={element.srcImage} alt={element.altText} />
          </S.GridElement>
        ))}
      </S.Grid>
    </S.Wrapper>
  </SectionBackground>
)

export default GridImage
