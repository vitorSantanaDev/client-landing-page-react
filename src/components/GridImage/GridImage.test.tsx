import React from 'react'
import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'

import GridImage from '.'
import mock from './mock'

describe('<GridImage />', () => {
  it('should render images', () => {
    const { container } = renderTheme(<GridImage {...mock} />)
    const images = screen.getAllByRole('img', { name: /teste/i })

    expect(images).toHaveLength(6)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should not render images when grid prop is empty', () => {
    const { container } = renderTheme(
      <GridImage title={mock.title} description="test" grid={[]} />
    )
    const images = screen.queryAllByRole('img', { name: /teste/i })

    expect(images).toHaveLength(0)
    expect(container.firstChild).toMatchSnapshot()
  })
})
