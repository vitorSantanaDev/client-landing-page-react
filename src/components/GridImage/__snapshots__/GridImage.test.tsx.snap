// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should not render images when grid prop is empty 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  width: 100%;
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  gap: 3.2rem;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div
  class="c0"
  id=""
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        My grid
      </h2>
      <div
        class="c4 c5"
      >
        test
      </div>
      <div
        class="c6"
      />
    </div>
  </div>
</div>
`;

exports[`<GridImage /> should render images 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  width: 100%;
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  gap: 3.2rem;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  max-height: 447px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div
  class="c0"
  id=""
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        My grid
      </h2>
      <div
        class="c4 c5"
      >
        Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <img
            alt="Teste 1"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=1"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 2"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=2"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 3"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=3"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 4"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=4"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 5"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=5"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Teste 6"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=6"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
